src/features/account/components/modals/AddToUserCircleModal.tsx

```diff
@@ -26,50 +28,77 @@ export const AddToUserCircleModal: FunctionComponent<
   );
   const dispatch = useDispatch();
 
-  const {
-    isLoading,
-    isVerified,
-    payload,
-    submitTypedSignRequest,
-    signatureError,
-  } = useTypedSignTrustCredential();
+  const [isLoading, setIsLoading] = useState(false);
+  const { issuerAddress, signMessage, signError, accountVCBuilder } =
+    useVerifiableCredential();
 
-  const { showErrorMsg, showSuccessMsg } = useToastMsg();
+  const { showSuccessMsg, showErrorMsg, showInfoMsg } = useToastMsg();
 
   const shortSubAddress = useMemo(
     () => trimAddress(subjectAddress),
     [subjectAddress],
   );
 
-  useEffect(() => {
-    if (signatureError) {
-      if (signatureError.type === SignatureErrorTypes.Error) {
-        showErrorMsg({
-          title: t`Failed to verify signature`,
-          description: signatureError.message,
-        });
-      } else {
-        showErrorMsg({
-          title: t`Invalid Signature`,
-          description: t`Your signature is invalid`,
-        });
-      }
+  const onSignButtonClick = useCallback(() => {
+    if (!issuerAddress) {
+      return;
+    }
+    if (!isAddress(subjectAddress)) {
+      return;
     }
-  }, [signatureError, showErrorMsg]);
+    setIsLoading(true);
+
+    const VC = accountVCBuilder.buildAccountTrust(
+      issuerAddress,
+      subjectAddress,
+    );
+
+    signMessage(VC)
+      .then((signature) => {
+        if (signature) {
+          const assertion = accountVCBuilder.getSignedAssertion(VC, signature);
+          console.log('assertion', assertion);
+
+          showSuccessMsg({
+            title: t`Added to your trust circle`,
+            description: t`${shortSubAddress} has been added to your trust circle`,
+          });
+
+          dispatch(addUserToUserCircle(subjectAddress));
+          dispatch(setAddToUserModalOpen(false));
+        } else {
+          showInfoMsg({
+            title: t`Info`,
+            description: t`The signature is empty, user deny signing`,
+          });
+        }
```

Comment:
you dont need that, as other module also not display this message

Reply:
Removed.